### PR TITLE
Truncate SNR/FAR for live pastro calcs

### PR DESCRIPTION
**Pull request by tdent**
This will require changes / additional info in the .json config file for the live p astro calculation, eg 
```
"method": "template_param_bins_types_farlim",
...
"limit_snr": 12.0,
"limit_far": 3.17e-10
```
for a limiting FAR of (just over) 1/100yr and SNR of 12.

### CODE REVIEW

File: pycbc/population/live_pastro_utils.py [hunk 1]
```diff
@@ -71,6 +76,24 @@ def __init__(self, specfile, bank):
...
+        if len(trigger_data['triggered']) > 1:
+            farlim = self.spec['limit_far']
+            snrlim = self.spec['limit_snr']
+            if trigger_data['far'] < farlim and \
+                    trigger_data['network_snr'] > snrlim:
+                logging.debug('Truncating FAR and SNR from %f, %f to %f, %f',
+                              trigger_data['far'], trigger_data['network_snr'],
+                              farlim, snrlim)
+                trigger_data['network_snr'] = snrlim
+                trigger_data['far'] = farlim
```
Comment (titodalcanton) [inline pycbc/population/live_pastro_utils.py:94]: My usual suggestion of negating these if statements, so as to return as early as possible and reduce indentation and nesting.

**Comment by titodalcanton** [comment]
One small comment but otherwise this looks good to me. Is there more coming? Should I test this at IJCLab?

**Comment by tdent** [comment]
Hopefully the return logic is still correct after 'negation', please check ..

File: pycbc/population/live_pastro_utils.py [hunk 2]
```diff
@@ -71,6 +76,32 @@ def __init__(self, specfile, bank):
...
+                return trigger_data
+
+            logging.debug('Truncating FAR and SNR from %f, %f to %f, %f',
+                          trigger_data['far'], trigger_data['network_snr'],
+                          farlim, snrlim)
+            trigger_data['network_snr'] = snrlim
+            trigger_data['far'] = farlim
+            return trigger_data
+
+        raise RuntimeError('Number of triggered ifos must be >0 !')
```
Comment (titodalcanton) [inline pycbc/population/live_pastro_utils.py:103]: I am pretty sure `len()` will not return a number less than zero, so this should look like this:
```suggestion
        farlim = self.spec['limit_far']
        snrlim = self.spec['limit_snr']
        # Only do anything if FAR and SNR are beyond given limits
        if trigger_data['far'] > farlim or \
                trigger_data['network_snr'] < snrlim:
            return trigger_data

        logging.debug('Truncating FAR and SNR from %f, %f to %f, %f',
                      trigger_data['far'], trigger_data['network_snr'],
                      farlim, snrlim)
        trigger_data['network_snr'] = snrlim
        trigger_data['far'] = farlim
        return trigger_data
```

Comment (tdent) [reply pycbc/population/live_pastro_utils.py:103]: No, the error is in the (unexpected) case where it is an empty list with length 0, which is not negative but also not >0.

**Comment by titodalcanton** [comment]
Testing at IJCLab.

**Review by titodalcanton — APPROVED** [review]

**Pull request merged**